lib/msf/core/exploit/http/client.rb

```diff
@@ -508,6 +508,31 @@ def path_from_uri(uri)
     end
   end
 
+  #
+  # Returns a hash of request opts from a URL string
+  def request_opts_from_url(url)
+    tgt = URI.parse(url)
+    opts = { 'rhost' => tgt.host, 'rport' => tgt.port, 'uri' => tgt.request_uri }
```

Comment:
You're entirely right, I should have pulled the full validation block you had going. Will adjust as requested, thx.